<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: Genotyping and QTL Mapping in DO Mice</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>
<h1> Genotyping and QTL Mapping in DO Mice
<img class="toplogo" src="../../../doc/html/logo.jpg" alt="[R logo]">
</h1>
<hr>
<div align="center">
<a href="../../../doc/html/packages.html"><img src="../../../doc/html/left.jpg" alt="[Up]" width="30" height="30" border="0"></a>
<a href="../../../doc/html/index.html"><img src="../../../doc/html/up.jpg" alt="[Top]" width="30" height="30" border="0"></a>
</div><h2>Documentation for package &lsquo;DOQTL&rsquo; version 0.99</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
</ul>

<h2>Help Pages</h2>


<table width="100%">
<tr><td width="25%"><a href="SNPtools-package.html">SNPmanipulation-package</a></td>
<td>Accessing, subsetting and plotting mouse SNPs.</td></tr>
<tr><td width="25%"><a href="argmax.geno.html">argmax.geno</a></td>
<td>Find the most probable genotype for all samples using the Viterbi algorithm.</td></tr>
<tr><td width="25%"><a href="batch.normalize.html">batch.normalize</a></td>
<td>Batch normalize the X &amp; Y intensity data.</td></tr>
<tr><td width="25%"><a href="bayesint.html">bayesint</a></td>
<td>Find a Bayesian Credible Interval around a QTL.</td></tr>
<tr><td width="25%"><a href="calc.genoprob.html">calc.genoprob</a></td>
<td>Calculate the 8 founder genotype probabilities at each SNP.</td></tr>
<tr><td width="25%"><a href="call.genotypes.html">call.genotypes</a></td>
<td>Call the most probably genotype by finding the Viterbi path through the genotype states for each sample.</td></tr>
<tr><td width="25%"><a href="categorize.variants.html">categorize.variants</a></td>
<td>categorize.variants</td></tr>
<tr><td width="25%"><a href="cluster.strains.html">cluster.strains</a></td>
<td>cluster.strains</td></tr>
<tr><td width="25%"><a href="convert.variants.to.GRanges.html">convert.variants.to.GRanges</a></td>
<td>convert.variants.to.GRanges</td></tr>
<tr><td width="25%"><a href="convert.variants.to.numeric.html">convert.variants.to.numeric</a></td>
<td>convert.variants.to.numeric</td></tr>
<tr><td width="25%"><a href="example.genes.html">example.genes</a></td>
<td>example.genes</td></tr>
<tr><td width="25%"><a href="example.snps.html">example.snps</a></td>
<td>example.snps</td></tr>
<tr><td width="25%"><a href="extract.raw.data.html">extract.raw.data</a></td>
<td>Extract intensities, genotypes and call rates from from raw MUGA data files</td></tr>
<tr><td width="25%"><a href="filter.geno.probs.html">filter.geno.probs</a></td>
<td>Remove SNPs where the genotype probabilities are too low for one founder state</td></tr>
<tr><td width="25%"><a href="filter.samples.html">filter.samples</a></td>
<td>Filter X, Y and genotype data by call rate</td></tr>
<tr><td width="25%"><a href="find.overlapping.genes.html">find.overlapping.genes</a></td>
<td>find.overlapping.genes</td></tr>
<tr><td width="25%"><a href="founder.cols.html">founder.cols</a></td>
<td>founder.cols</td></tr>
<tr><td width="25%"><a href="gene.plot.html">gene.plot</a></td>
<td>gene.plot</td></tr>
<tr><td width="25%"><a href="get.mgi.features.html">get.mgi.features</a></td>
<td>get.mgi.features</td></tr>
<tr><td width="25%"><a href="get.pattern.variants.html">get.pattern.variants</a></td>
<td>get.pattern.variants</td></tr>
<tr><td width="25%"><a href="get.strains.html">get.strains</a></td>
<td>get.strains</td></tr>
<tr><td width="25%"><a href="get.variants.html">get.variants</a></td>
<td>get.variants</td></tr>
<tr><td width="25%"><a href="kinship.html">kinship</a></td>
<td>Create a kinship matrix based on allele sharing.</td></tr>
<tr><td width="25%"><a href="plot.haplotype.blocks.html">plot.haplotype.blocks</a></td>
<td>Given a genomic region, plot the haplotype blocks for a subset of strains.</td></tr>
<tr><td width="25%"><a href="predict.sex.html">predict.sex</a></td>
<td>Determine the sex of each sample</td></tr>
<tr><td width="25%"><a href="qtl.html">qtl</a></td>
<td>qtl</td></tr>
<tr><td width="25%"><a href="qtl.heatmap.html">qtl.heatmap</a></td>
<td>Plot a Heatmap of all QTL</td></tr>
<tr><td width="25%"><a href="qtl.qtlrel.html">qtl.qtlrel</a></td>
<td>Use QTLRel to map a set of traits</td></tr>
<tr><td width="25%"><a href="read.cross.html">read.cross</a></td>
<td>Read in the data to genotype a DO cross.</td></tr>
<tr><td width="25%"><a href="sex.founders.html">sex.founders</a></td>
<td>Data.frame with founder IDs and sex value.</td></tr>
<tr><td width="25%"><a href="snp.plot.html">snp.plot</a></td>
<td>snp.plot</td></tr>
<tr><td width="25%"><a href="SNPtools-package.html">SNPmanipulation</a></td>
<td>Accessing, subsetting and plotting mouse SNPs.</td></tr>
<tr><td width="25%"><a href="states.html">states</a></td>
<td>states</td></tr>
<tr><td width="25%"><a href="summarize.genotype.data.html">summarize.genotype.data</a></td>
<td>Summarize the genotype data output by the genotyping HMM.</td></tr>
<tr><td width="25%"><a href="variant.plot.html">variant.plot</a></td>
<td>variant.plot</td></tr>
</table>
</body></html>
